### PR TITLE
Clarity on declarative and imperative approaches with Pulumi

### PR DESCRIPTION
**Pull request by interurban**
Adding a section on our how Pulumi works to describe Pulumi's powerful declarative model and how this relates to using imperative languages to author y our solutions.

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

**Comment by interurban** [comment]
Hey @ringods I've made some adjustments to your blog title to help with SEO and would appreciate your thoughts on how i've incorporated things into the docs. thanks!

File: themes/default/content/docs/concepts/how-pulumi-works.md [hunk 1]
```diff
@@ -17,35 +17,37 @@ aliases:
...
 
 The following diagram illustrates the interaction between these parts of the system:
 
-<img src="/images/docs/reference/engine-block-diagram.png" alt="Pulumi engine and providers" width="600">
+<img src="/images/docs/reference/engine-block-diagram.png" alt="Pulumi IaC system architecture, the Pulumi engine and providers" width="600">
 
 In the next section, we will describe each of these components and see how they all fit together during an invocation of `pulumi up`.
 
-## Language Hosts
+## Language hosts
```
Comment (ringods) [inline themes/default/content/docs/concepts/how-pulumi-works.md:30]: The second word is made lowercase here, but not in other titles further down like `Resource Providers`. Is this an inconsistency now, or is there still a reason behind it? I'm not an native English speaker, so I might not know.

File: themes/default/content/docs/concepts/how-pulumi-works.md [hunk 2]
```diff
@@ -17,35 +17,37 @@ aliases:
...
 2. An _SDK_ which provides bindings for each type of resource the provider can manage.
 
-Like the language runtime itself, the SDKs are available as regular packages.  For example, there is a [`@pulumi/aws`](https://www.npmjs.com/package/@pulumi/aws) package for Node available on npm and a [`pulumi_aws`](https://pypi.org/project/pulumi-aws) package for Python available on PyPI.  When these packages are added to your project, they run [`pulumi plugin install`](/docs/cli/commands/pulumi_plugin_install)  behind the scenes in order to download the resource plugin from pulumi.com.
+Like the language runtime itself, the SDKs are available as regular packages. For example, there is a [`@pulumi/aws`](https://www.npmjs.com/package/@pulumi/aws) package for Node available on npm and a [`pulumi_aws`](https://pypi.org/project/pulumi-aws) package for Python available on PyPI.  When these packages are added to your project, they run [`pulumi plugin install`](/docs/cli/commands/pulumi_plugin_install) behind the scenes to download the resource plugin from Pulumi.com.
```
Comment (ringods) [inline themes/default/content/docs/concepts/how-pulumi-works.md:50]: > ... download the resource plugin from Pulumi.com or partners.

We do not maintain all the SDKs or plugin binaries.

**Review by ringods — APPROVED** [review]

**Review by scottslowe — CHANGES_REQUESTED** [review]

**Review by scottslowe — APPROVED** [review]

File: themes/default/content/blog/pulumi-is-imperative-declarative-imperative/index.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "Pulumi Is Imperative, Declarative, and Imperative"
+title: "Pulumi's Declarative and Imperative Approach IaC"
```
Comment (cnunciato) [inline themes/default/content/blog/pulumi-is-imperative-declarative-imperative/index.md:2]: > Pulumi's Declarative and Imperative Approach to IaC"

Missing a "to".

We can also add `allow_long_title: true` to get past the 60-character link check if we need to.

File: themes/default/content/docs/concepts/how-pulumi-works.md [hunk 4]
```diff
@@ -383,10 +385,20 @@ resources:
...
+
+## Declarative and imperative approach
+
+With Pulumi, you author your infrastructure in your preferred programming language. When you run `pulumi up`, the Pulumi CLI starts both language host for your selected programming language as well as the required providers (via the Pulumi engine). The providers, coordinated by the Pulumi engine, are the ones that perform the actual creation, modification, or deletion of your infrastructure. The separation of language support from the engine is what makes Pulumi so powerful, providing the best of both imperative and declarative approaches for your infrastructure as code solutions.
```
Comment (cnunciato) [inline themes/default/content/docs/concepts/how-pulumi-works.md:398]: This is all correct, but it feels a bit like much of what's here repeats what's already been mentioned above and also leaves out "the meat" of what this is presumably about, namely explaining "how Pulumi is declarative" -- assuming that's what we were aiming for. If we want to keep from duplicating content, would it make sense to quote, say, a portion of Ringo's post here, and link to the rest? Or pull more of the content in and rework it somehow?

**Review by cnunciato — COMMENTED** [review]
Sorry for not getting this in sooner? It merged while I was still in the middle of my review. Feel free to use what you like. Thanks!

**Pull request merged**